src/operation/mod.rs

```diff
@@ -57,6 +65,12 @@ pub(crate) trait Operation {
     /// Interprets the server response to the command.
     fn handle_response(&self, response: CommandResponse) -> Result<Self::O>;
 
+    /// Interpret an error encountered while sending the built command to the server, potentially
+    /// recovering.
+    fn handle_error(&self, error: Error) -> Result<Self::O> {
```

Comment:
Ironically, this amendment to `Operation` is similar to the `handles_command_errors` that I just removed, but I think it's a more general method that may prove to be useful in other cases in the future.